doc/040_backup.rst

```diff
@@ -114,24 +114,59 @@ restic encounters:
-time/date), restic always needs to read all files and only afterwards
-can compute which parts of the files need to be saved. When you backup
-the same directory again (maybe with new or changed files) restic will
-find the old snapshot in the repo and by default only reads those files
-that are new or have been modified since the last snapshot. This is
-decided based on the following attributes of the file in the file system:
-
- * Type (file, symlink, or directory?)
- * Modification time
- * Size
- * Inode number (internal number used to reference a file in a file system)
-
 Now is a good time to run ``restic check`` to verify that all data
 is properly stored in the repository. You should run this command regularly
 to make sure the internal structure of the repository is free of errors.
 
+File change detection
+*********************
+
+restic uses cryptographic hashes of file contents to deduplicate data both
+within and across backups. But, since hashing files is expensive, it also
+uses a heuristic based on file metadata to determine whether a file is likely
+unchanged since a previous backup. If it is, the file is not hashed again.
+
+The change detection only runs on regular files (not special files, symlinks
+or directories) that have the exact same path as they did in a previous
+backup of the same location.  If a file or one of its containing directories
+was renamed, it is considered a different file and its entire contents will
+be scanned.
+
+On **Unix** (including Linux and Mac), given that a file lives at the same
+location as a file in a previous backup, the following file metadata
+attributes have to match for its contents to be presumed unchanged:
+
+ * Modification timestamp (mtime).
+ * Metadata change timestamp (ctime).
+ * File size.
+ * Inode number (internal number used to reference a file in a filesystem).
+
+The reason for requiring both mtime and ctime to match is that Unix programs
+can freely change these timestamps (and some do), so that neither is fully
+reliable. Checking both timestamps is a compromise that errs on the side of
+caution.
+
+The following ``restic backup`` command line flags modify the change detection
+rules:
+
+ * ``--force``: turn off change detection and rescan all files.
+ * ``--trust-mtime``: require mtime to match, but allow ctime to differ.
+ * ``--ignore-inode``: require mtime to match, but allow inode number
+   and ctime to differ.
+
+The option ``--ignore-inode`` exists to support FUSE-based filesystems and
+pCloud, which do not assign stable inodes to files.
+
+Note that the device id of the containing mount point is not taken into
+account, because it very seldom changes.
+
+On **Windows**, a file is considered unchanged when its path and modification
+time match, and only ``--force`` has any effect. The other options are
+recognized but ignored.
+
+All the above pertains only to file contents. Metadata changes (permissions,
```

Comment:
I just stumbled over the word `pertains` which I didn't know before. Its meaning is deductible from the context, but I'd prefer to replace it with something simpler. Maybe `Metadata changes (...) are always included in the backup even if the file content is considered unchanged.`

Reply:
Done, and moved this up, so it's not mistaken for a Windows-only feature.